localstack-core/localstack/services/sqs/provider.py

```diff
@@ -507,6 +507,7 @@ def _run(self, move_task: MessageMoveTask):
 
                 target_queue.put(
                     message=message.message,
+                    context=move_task.submitted_request_context,
```

Comment:
The `MessageMoveTask` now stores the submittor `StartMessageMoveTask` request's context. So just extract it from and pass it around.